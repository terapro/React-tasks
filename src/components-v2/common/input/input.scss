.search-input {
  width: $appWidth - 2 * $elementPadding;
  background-color: #000;
  border: 0px;
  border-bottom: 2px solid $color_red;
  height: 45px;
  color: #EAEAEA;
  padding: 0 $searchInputPaddingHorizontal;
  font-size: 20px;
  line-height: 45px;
}
