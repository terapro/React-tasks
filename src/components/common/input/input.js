import React from 'react';
import PropTypes from 'prop-types';



const Input = (props) => (

  <input className={'search-input'} type={'text'} placeholder = 'Start typing...' onInput = {props.searchInputCallback} value = {props.value} onKeyPress={props.onKeyPressedFunc} />
);


Input.propTypes = {
  placeholder: PropTypes.string,
  searchInputCallback: PropTypes.func.isRequired,
  value: PropTypes.string,
  onKeyPressedFunc: PropTypes.func
};

Input.defaultProps = {
  placeholder: '',
  searchInputCallback: () => {},
  value: '',
  onKeyPressedFunc: () => {}
};
export {Input};