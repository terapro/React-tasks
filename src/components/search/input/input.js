import React from 'react';
import PropTypes from 'prop-types';

const Input = (props) => (
    <input className={'search-input'} type={'text'} placeholder={props.placeholder}/>
);

Input.propTypes = {
    placeholder: PropTypes.string
};


export {Input};