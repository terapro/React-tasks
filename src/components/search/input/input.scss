.search-input {
  width: $headerContainerWidth - 2 * $searchInputPaddingHorizontal;
  background-color: #000;
  border: 0px;
  border-bottom: 2px solid $color_red;
  height: 45px;
  color: #EAEAEA;
  padding: 0 $searchInputPaddingHorizontal;
  font-size: 20px;
  line-height: 45px;
}