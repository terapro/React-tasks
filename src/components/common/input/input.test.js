import React from 'react';
import {Input} from 'src/components/common/input/input.js';
import {shallow} from 'enzyme';

const mockFn = jest.fn();
const mockText = 'testing Input..';

describe('<Input />', () => {
  describe('Rendering', () => {
    it('should render the component', () => {
      const wrapper = shallow(<Input
        searchInputCallback = {mockFn}
        onKeyPressedFunc = {mockText}
      />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});