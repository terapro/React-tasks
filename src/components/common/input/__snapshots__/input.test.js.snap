// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Rendering should render the component 1`] = `
<input
  className="search-input"
  onInput={[MockFunction]}
  onKeyPress="testing Input.."
  placeholder="Start typing..."
  type="text"
  value=""
/>
`;

exports[`<Input /> should render the input 1`] = `
<input
  className="search-input"
  onInput={[Function]}
  onKeyPress={[Function]}
  placeholder="Start typing..."
  type="text"
  value="testing Input.."
/>
`;
